### ISSUE TITLE
Windows with enforced DEP doesn't support emulation with dynarac

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jennorby/mupen64plus#361]
```
rehack reported via irc that he wanted to run his emulator in windows xp
64bit sp2 with enforced DEP (NX memory protection) for all programs. The
emulator just stops without further notice.

He whitelisted m64p and it just ran afterwards.

Solution is to write malloc_exec for windows. An example was given by him:

void *malloc_exec(size_t size)
{
 return VirtualAlloc(NULL, size, MEM_COMMIT | MEM_RESERVE,
PAGE_EXECUTE_READWRITE);
}

He also said that a free function must be implemented (but maybe also
works with free...)

VirtualFree(address, 0, MEM_RELEASE)

```

Original issue reported on code.google.com by `sven@narfation.org` on 4 Apr 2010 at 7:14

**Comment by GoogleCodeExporter** [comment]
```
free(...) to change to free_exec are in

 * realloc_exec
 * interupt.c: if (blocks[i]->block) { free(blocks[i]->block); blocks[i]->block =
NULL; }
 * r4300.c: if (blocks[i]->block) { free(blocks[i]->block); blocks[i]->block = NULL;
}

I don't have something to test it - so I just assume that this is correct.
```

Original comment by `sven@narfation.org` on 4 Apr 2010 at 7:24

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I think it should be fixed by 6bb44b4ffdd4
```

Original comment by `sven@narfation.org` on 6 Apr 2010 at 8:03

**Comment by GoogleCodeExporter** [comment]
Original comment by `richard...@gmail.com` on 10 Apr 2010 at 1:36
- Changed state: **Fixed**
